### PR TITLE
add semigroup instances

### PR DESCRIPTION
**Pull request by fumieval**

### CODE REVIEW

File: euphoria.cabal [hunk 1]
```diff
@@ -97,6 +97,7 @@ library
                      , containers >= 0.5.5
                      , unordered-containers >= 0.2.5
                      , transformers >= 0.4.1
+                     , semigroups
```
Comment (maoe) [inline euphoria.cabal]: Should we apply https://ghc.haskell.org/trac/ghc/wiki/Migration/8.4#base-4.11.0.0?

**Review by maoe — CHANGES_REQUESTED** [review]
I think this patch still fails with GHC 8.6. You can rebase it onto #25 to see which versions of GHC fail.

I got this with 8.6.2:
```
[2 of 4] Compiling Data.EnumSet     ( Data/EnumSet.hs, dist/build/Data/EnumSet.o )

Data/EnumSet.hs:24:15: error:
    • No instance for (GHC.Base.Semigroup (EnumSet k))
        arising from the 'deriving' clause of a data type declaration
      Possible fix:
        use a standalone 'deriving instance' declaration,
          so you can specify the instance context yourself
    • When deriving the instance for (Monoid (EnumSet k))
   |
24 |     (Eq, Ord, Monoid, Typeable, Data, NFData)
   |               ^^^^^^
cabal: Failed to build enummapset-th-0.6.1.1 (which is required by
euphoria-0.8.0.0). See the build log above for details.
```

File: euphoria.cabal [hunk 2]
```diff
@@ -67,8 +67,9 @@ category:            FRP
 build-type:          Simple
 cabal-version:       >=1.8
 homepage:            http://github.com/tsurucapital/euphoria
-tested-with:         GHC == 7.10.3
-                   , GHC == 8.0.1
+tested-with:         GHC == 8.6.4
+                   , GHC == 8.8.3
+                   , GHC == 8.10.3
```
Comment (maoe) [inline euphoria.cabal:72]: 8.10.1?

Comment (fumieval) [reply euphoria.cabal]: it was some mirage...

**Review by maoe — APPROVED** [review]

**Comment by maoe** [comment]
LGTM

**Pull request merged**